### PR TITLE
correct float literals, 32/64 trunc, unref vars

### PR DESCRIPTION
**Pull request by diablodale**
The depthai-core portion of fixes for luxonis/depthai-core#248

Together with the already merged depthai-shared portion, they eliminate hundreds of warnings to lessen noise.

### CODE REVIEW

**Review by szabi-luxonis — APPROVED** [review]

**Review by szabi-luxonis — CHANGES_REQUESTED** [review]
CI is failing due to style checks.
`clangformat` target should solve it, but I assume you don't have it installed on windows?

**Comment by diablodale** [comment]
It is enabled in cmake, but such isn't available in typical Windows installations so cmake it likely quietly ignoring that target.

Alternatively, I used the VSCode c++ extension from Microsoft and have enabled. It does apply clang-format when I type a semicolon. But in many of these, I never typed a semi...I just inserted the casts inside the line. So it never formatted.
I del/add a semi on one of the lines lang-format reported and it reformatted it in the same way as the CI report.

I've turned on the additional option to clang-format on save of the file. Makes me a bit paranoid because I'll likely not notice the reformatting, but I get your team's need for this and I can work with that.
I'll resave the two files that the CI reported wrong and force push.

That all ok with u?

**Comment by szabi-luxonis** [comment]
> It is enabled in cmake, but such isn't available in typical Windows installations so cmake it likely quietly ignoring that target.
> 
> Alternatively, I used the VSCode c++ extension from Microsoft and have enabled. It does apply clang-format when I type a semicolon. But in many of these, I never typed a semi...I just inserted the casts inside the line. So it never formatted. I del/add a semi on one of the lines lang-format reported and it reformatted it in the same way as the CI report.
> 
> I've turned on the additional option to clang-format on save of the file. Makes me a bit paranoid because I'll likely not notice the reformatting, but I get your team's need for this and I can work with that. I'll resave the two files that the CI reported wrong and force push.
> 
> That all ok with u?

Yes, that's fine.

**Comment by diablodale** [comment]
🤔 the CI here isn't getting the forcepush I did. Its available at https://github.com/diablodale/depthai-core/tree/fix248-trunc-2 but it isn't showing up on the tabs here (and therefore not reprocessing it)

**Comment by themarpe** [comment]
Reran the checks - will merge afterwards :+1:

File: src/utility/matrixOps.hpp [hunk 1]
```diff
@@ -136,8 +136,8 @@ bool matInv(std::vector<std::vector<float>>& A, std::vector<std::vector<float>>&
 
     std::vector<float> temp;
     // Find Inverse using formula "inverse(A) = adj(A)/det(A)"
-    for(size_t i = 0; i < A.size(); i++) {
-        for(size_t j = 0; j < A.size(); j++) {
+    for(size_t i = 0; i < A.size(); ++i) {
```
Comment (alex-luxonis) [inline src/utility/matrixOps.hpp:139]: Just curious on the `i++` -> `++i` changes, were there any warnings triggered by the previous version, or it's more like a style preference?

Comment (szabi-luxonis) [reply src/utility/matrixOps.hpp:139]: Porbably style, pre-increment is faster than post-increment, theoretically, but not for POD.

Comment (diablodale) [reply src/utility/matrixOps.hpp:139]: pre is my preferred for best opportunity for optimization, see notes https://en.cppreference.com/w/cpp/language/operator_incdec
No guarantee compilers old/new on all platforms won't use the slow path on POD. So I prefer pre.

I did not do it everywhere i that matrix code as some places need the post behavior.

FYI, the Eigen library is fantastically excellent and all header. Even for light use like cali processing. They've been testing  and optimizing across years and more platforms than most anyone could redo.

**Pull request merged**